File: dotnet/GHIElectronics.Due/Infrared.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GHIElectronics.DUE {
    public partial class DUEController {
        public class InfraredController {

            SerialInterface serialPort;

            public InfraredController(SerialInterface serialPort) => this.serialPort = serialPort;

            public int Read() {
                var cmd = "print(irread())";
                this.serialPort.WriteCommand(cmd);

                var res = this.serialPort.ReadRespone();
                var val = 255;
                if (res.success) {
                   

                    try {
                        val = int.Parse(res.respone);
                    }
                    catch { }
                }

                return val;

            }

            //public bool Clear() {
            //    var cmd = "R" + " " + "C";

            //    this.serialPort.WriteLine(cmd);

            //    var res = this.serialPort.ReadRespone();

            //    if (res.success) {
            //        return true;
            //    }

            //    return false;
            //}

            public bool Enable(bool enable) {
                var cmd = string.Format("irenable({0})", enable == true?1:0);

                this.serialPort.WriteCommand(cmd);

                var res = this.serialPort.ReadRespone();

                if (res.success) {
                    return true;
                }

                return false;
            }

        }
    }
}

File: micropython/test/duelink/infrared.py
class InfraredController:
    def __init__(self, serialPort):
        self.serialPort = serialPort

    def Read(self)->int:
        cmd = "irread()"
        self.serialPort.WriteCommand(cmd)
        r,s = self.serialPort.ReadResponse()

        if r:
            try:
                return int(s)
            except:
                pass
        return -1
    
    def Write(self, command: int)->bool:
        cmd = f"IrWrite({command})"
        self.serialPort.WriteCommand(cmd)
        r,s = self.serialPort.ReadResponse()
        return r

    def Enable(self, txpin:int, rxpin: int)->bool:
        cmd = f"iren({txpin}, {rxpin}})"
        self.serialPort.WriteCommand(cmd)

        r,s = self.serialPort.ReadResponse()
        return r
